src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs

```diff
@@ -92,7 +92,30 @@ private ClientConfigPaths(string exePath, bool includeUserConfig)
                 }
             }
 
-            if (!string.IsNullOrEmpty(ApplicationUri))
+            string externalConfigPath = AppDomain.CurrentDomain.GetData("APP_CONFIG_FILE") as string;
+            if (!string.IsNullOrEmpty(externalConfigPath))
+            {
+                if (Uri.IsWellFormedUriString(externalConfigPath, UriKind.Absolute))
+                {
+                    Uri externalConfigUri = new Uri(externalConfigPath);
+                    if (externalConfigUri.IsFile)
+                    {
+                        ApplicationConfigUri = externalConfigUri.LocalPath;
+                    }
+                }
+                else
+                {
+                    // Uri constructor would throw for relative paths but full framework doesn't.
+                    // We will mimic full framework behavior here.
+
+                    // If path starts with directory separator Path.Combine would combine with a root of BaseDirectory so we trim that character to avoid that
+                    // For absolute path we should already be covered by the other code path
```

Comment:
I do not think that this is a correct statement for Linux.

`Uri.IsWellFormedUriString("/home/jkotas/myconfig.xml", UriKind.Absolute)` returns false on Linux.

Reply:
you're right, I've done some combination of your solution but which supports URIs as well